aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -80,15 +80,28 @@ static inline ScalarType combine_categories(ScalarType higher, ScalarType lower)
   // NOLINTNEXTLINE(bugprone-branch-clone)
   if(isComplexType(higher)) {
     return higher;
-  }
-  else if(!isComplexType(lower) && isFloatingType(higher)) {
+  } else if (isComplexType(lower)) {
+    // special case for bfloat16
+    if (higher == kBFloat16) {
+      // BFloat16 corresponds to Float in terms of range,
+      // so we return ComplexFloat
+      return kComplexFloat;
```

Comment:
what if lower if complexhalf? (I think the expected out dtype is complexhalf in that case)

Reply:
```python
import torch
from torch._prims.utils import elementwise_dtypes, ELEMENTWISE_TYPE_PROMOTION_KIND

s = torch.tensor(1j, dtype=torch.chalf)
t = torch.zeros(3, dtype=torch.bfloat16)

print(elementwise_dtypes(s, t, type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT))
```

Output
```
(torch.complex64, torch.complex64)
```